File: update_india.js
const fs = require('fs');
const jsdom = require('jsdom');
const { JSDOM } = jsdom;

// Read site_3.txt
const html = fs.readFileSync('site_3.txt', 'utf-8');
const dom = new JSDOM(html);
const document = dom.window.document;

// Grab all selects and their options
const selects = document.querySelectorAll('select');
const optionsMap = {};

selects.forEach(sel => {
    const name = sel.getAttribute('name') || sel.getAttribute('id');
    if (!name) return;
    
    const options = [];
    sel.querySelectorAll('option').forEach(opt => {
        const txt = opt.textContent.trim();
        const lower = txt.toLowerCase();
        if (txt && !['select', '---select---', 'select one', '-select-', 'choose'].includes(lower)) {
            options.push(txt);
        }
    });
    
    if (options.length > 0) {
        optionsMap[name] = options;
    }
});

// Read indian_visa_config.js
let content = fs.readFileSync('modules/indian_visa_config.js', 'utf-8');

// The file likely exports a const object. Let's do a trick using eval to parse it.
const regex = /export const INDIAN_VISA_FIELDS = (\{[\s\S]*?\});(?:\n|$)/m;
const match = content.match(regex);

if (match) {
    let objStr = match[1];
    let fields;
    eval(`fields = ${objStr};`);
    
    // Process fields
    for (const [key, data] of Object.entries(fields)) {
        const htmlName = data.name;
        if (optionsMap[htmlName]) {
            const opts = optionsMap[htmlName];
            let optStr = " [" + opts.join(', ') + "]";
            if (optStr.length > 200) {
                optStr = " [" + opts.slice(0, 5).join(', ') + ", ...]";
            }
            // Remove existing brackets if present and append new ones
            data.label = data.label.split(' [')[0] + optStr;
        }
    }
    
    // Stringify back
    const newFieldsJson = JSON.stringify(fields, null, 4);
    content = content.replace(regex, `export const INDIAN_VISA_FIELDS = ${newFieldsJson};`);
    
    fs.writeFileSync('modules/indian_visa_config.js', content, 'utf-8');
    console.log("Updated INDIAN_VISA_FIELDS successfully.");
} else {
    console.log("Could not find INDIAN_VISA_FIELDS. Using backup regex...");
}

File: update_india.py
import json
import bs4
import re

html = open('site_3.txt', encoding='utf-8').read()
soup = bs4.BeautifulSoup(html, 'html.parser')

selects = soup.find_all('select')

options_map = {}
for sel in selects:
    name = sel.get('name')
    if not name:
        name = sel.get('id')
    if not name: continue
    
    options = []
    for opt in sel.find_all('option'):
        txt = opt.text.strip()
        if txt and txt.lower() not in ['select', '---select---', 'select one', '-select-', 'choose']:
            options.append(txt)
    if options:
        options_map[name] = options

content = open('modules/indian_visa_config.js', encoding='utf-8').read()

# Extract INDIAN_VISA_FIELDS
match = re.search(r'export const INDIAN_VISA_FIELDS = (\{.*?\});\n', content, re.DOTALL)
if match:
    fields_json = match.group(1)
    # The JSON might have some trailing commas or comments, but let's assume it's clean enough or use a dirty json loader.
    # Actually, ast.literal_eval or json won't work perfectly if it's strictly JS. We'll try json.loads.
    try:
        fields = json.loads(fields_json)
        for key, data in fields.items():
            html_name = data.get('name', '')
            if html_name in options_map:
                opts = options_map[html_name]
                opt_str = " [" + ", ".join(opts) + "]"
                if len(opt_str) > 200:
                    opt_str = " [" + ", ".join(opts[:5]) + ", ...]"
                data['label'] = data['label'].split(' [')[0] + opt_str

        new_fields_json = json.dumps(fields, indent=4)
        new_content = content[:match.start(1)] + new_fields_json + content[match.end(1):]
        open('modules/indian_visa_config.js', 'w', encoding='utf-8').write(new_content)
        print("Updated INDIAN_VISA_FIELDS successfully.")
    except Exception as e:
        print("Failed to parse JSON:", e)
else:
    print("Could not find INDIAN_VISA_FIELDS.")
